### PR TITLE
Helm integration

### PR DESCRIPTION
**Pull request by tamas-molnar**
Helm & Jenkins integration - doesn't affect functionality.

- vendoring
- circleCI updates
- healthcheck updates

### CODE REVIEW

**Comment by tamas-molnar** [comment]
will be decommed

**Comment by tamas-molnar** [comment]
suprise!

File: authors/handlers.go [hunk 1]
```diff
@@ -89,30 +89,46 @@ func (h *AuthorHandler) GetCount(writer http.ResponseWriter, req *http.Request)
...
+func (h *AuthorHandler) HealthCheck() fthealth.Check {
+	return fthealth.Check{
 		BusinessImpact:   "Unable to respond to requests",
-		Name:             "Check service has finished initilising.",
-		PanicGuide:       "https://sites.google.com/a/ft.com/ft-technology-service-transition/home/run-book-library/v1-authors-transformer",
+		Name:             "Check service has finished initialising.",
+		PanicGuide:       "TBD",
 		Severity:         1,
 		TechnicalSummary: "Cannot serve any content as data not loaded.",
 		Checker: func() (string, error) {
```
Comment (MMoisa) [inline authors/handlers.go:99]: Shouldn't we call the method `initChecker` here?

File: authors/handlers_test.go [hunk 2]
```diff
@@ -381,8 +382,19 @@ func router(s AuthorService) *mux.Router {
...
+			SystemCode:  "v1-authors-tf",
+			Name:        "V1 Authors Transformer",
+			Description: "It pulls and transforms V1/TME Authors into the UPP JSON model of an Author.",
+			Checks:      []fthealth.Check{h.HealthCheck()},
+		},
+		Timeout: 10 * time.Second,
+	}
+
+	m.HandleFunc("/__health", fthealth.Handler(timedHC))
+	//m.HandleFunc(status.GTGPath, status.NewGoodToGoHandler(h.GTG))
```
Comment (MMoisa) [inline authors/handlers_test.go]: Remove this line

File: circle.yml [hunk 3]
```diff
@@ -1,17 +1,30 @@
+machine:
```
Comment (MMoisa) [inline circle.yml]: I think we should upgrade this to circle2

File: helm/v1-authors-transformer/values.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+service:
+  name: "" # The name of the service, should be defined in the specific app-configs folder.
+  hasHealthcheck: "true"
+replicaCount: 1
+image:
+  repository: coco/v1-authors-transformer
+  pullPolicy: IfNotPresent
+resources:
+  limits:
+    memory: 1Gi
```
Comment (MMoisa) [inline helm/v1-authors-transformer/values.yaml]: Does the memory limit really need to be 1Gi? According to Grafana average memory load is below 20Mi. Even when there were a few cpu usage spikes there hasn't been much of an increase in memory usage.

File: authors/handlers_test.go [hunk 5]
```diff
@@ -13,12 +13,13 @@ import (
 	"sync"
```
Comment (MMoisa) [inline authors/handlers_test.go:13]: Make sure that the imports are properly formatted in the whole project

File: authors/handlers.go (hunk 1 above)
Comment (tamas-molnar) [reply authors/handlers.go:99]: Well the behavior of `__gtg` and `__health` wasn't the same before, maybe for a reason?

File: helm/v1-authors-transformer/values.yaml (hunk 4 above)
Comment (tamas-molnar) [reply helm/v1-authors-transformer/values.yaml]: Yeah I had a look at this too, that's why I set the request low - the top limit is just to determine when it's killed with an OOM

**Comment by MMoisa** [comment]
[Migrate v1-authors-transformer](https://trello.com/c/f8sdPPf9/108-migrate-v1-authors-transformer)

**Review by nwrigh — APPROVED** [review]

**Pull request merged**